jira/client.py

```diff
@@ -514,16 +514,15 @@ def __del__(self):
     def close(self):
         session = getattr(self, "_session", None)
         if session is not None:
+            try:
+                session.close()
+            except TypeError:
+                # TypeError: "'NoneType' object is not callable"
+                # Could still happen here because other references are also
+                # in the process to be torn down, see warning section in
+                # https://docs.python.org/2/reference/datamodel.html#object.__del__
+                pass
             self._session = None
-            if self.sys_version_info < (3, 4, 0):  # workaround for https://github.com/kennethreitz/requests/issues/2303
```

Comment:
I see that you want too call this code for all versions of Python. Are you sure this is needed? I don't see how this change fits with the others.

Reply:
I tested on Python 2.7, 3.6 and 3.7. And this condition  was the source of the socket unclosed. Also tests are OK.
So I think it is safe to say that this code is OK for all Python version :)